### PR TITLE
feat(dev ex): Update Lemon Button with new color design system

### PR DESCRIPTION
**Pull request by adamleithp**
## Problem
Our codebase has an abundance of random color CSS variables, and the inititive is to reduce the number of variables to allow engineers to move faster without so much guesswork.

## Changes
LemonButton.scss is now using our "3030" semantic CSS color variables.

## Does this work well for both Cloud and self-hosted?
Yes

## How did you test this code?
I've tested the buttons throughout our dashboard and in storybook.
I've gone in as far as I can in each product/page to try and catch edge cases.

### CODE REVIEW

File: frontend/__snapshots__/scenes-other-password-reset--initial--dark.png
Comment (daibhin) [inline frontend/__snapshots__/scenes-other-password-reset--initial--dark.png:1]: You'll need to be careful with the logged out pages. Growth were concerned about affecting conversion on the signup page in particular. At the time we opted to always leave them in light mode so as not to affect anything

File: frontend/__snapshots__/components-activitylog--data-management-activity--dark.png
Comment (daibhin) [inline frontend/__snapshots__/components-activitylog--data-management-activity--dark.png:1]: Is the loss of the background intentional?

File: frontend/__snapshots__/scenes-other-password-reset--initial--dark.png
Comment (adamleithp) [reply frontend/__snapshots__/scenes-other-password-reset--initial--dark.png:1]: I'll absolutely keep that in mind and I'll address this!

File: frontend/__snapshots__/components-activitylog--data-management-activity--dark.png
Comment (adamleithp) [reply frontend/__snapshots__/components-activitylog--data-management-activity--dark.png:1]: This is so strange! All the screenshots have no background!

**Comment by adamleithp** [comment]
Closing due to changes, which suggest another PR altogether.

File: frontend/__snapshots__/components-activitylog--data-management-activity--dark.png
Comment (adamleithp) [reply frontend/__snapshots__/components-activitylog--data-management-activity--dark.png:1]: This issue was due to a missing semicolon in the CSS. Closed PR in favour of another with less changes/history, that will not move forward in the end.

**Pull request closed without merging**